### PR TITLE
104 hardcoded tag list

### LINKED ISSUE #104: Hardcode tag list for post content
Add in hardcoded tags for the posts and then add in filter support for tags.

### PR DESCRIPTION
**Pull request by MarshallAsch**
closes #104. Added in a fixed tag list instead of a free text field to add a bit more consistency and structure. The actual list of tags needs to be figured out still.

### CODE REVIEW

File: app/src/main/java/ca/marshallasch/veil/tagList/TagListAdapter.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+     * This class will hold the view view for the cell.
+     */
+    private class ViewHolder {
+        private TextView textView;
+        private CheckBox checkBox;
+    }
+
+    /**
+     * Constructor with the default list of tags.
+     * @param context the application context
```
Comment (weihanli101) [inline app/src/main/java/ca/marshallasch/veil/tagList/TagListAdapter.java]: change "application" in `@param context the application context` to "activity's."

File: app/src/main/java/ca/marshallasch/veil/tagList/TagListAdapter.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        for (String tagOption : tagOptions) {
+            tagList.add(new ListState(tagOption, false));
+        }
+
+        addAll(tagList);
+    }
+
+    /**
+     * Pass in the list of tags to use
+     * @param context the application context
```
Comment (weihanli101) [inline app/src/main/java/ca/marshallasch/veil/tagList/TagListAdapter.java]: change "application"  to "activity's."

File: app/src/main/java/ca/marshallasch/veil/tagList/TagListAdapter.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    private class ViewHolder {
+        private TextView textView;
+        private CheckBox checkBox;
+    }
+
+    /**
+     * Constructor with the default list of tags.
+     * @param context the application context
+     */
+    public TagListAdapter(@NonNull Context context)
```
Comment (weihanli101) [inline app/src/main/java/ca/marshallasch/veil/tagList/TagListAdapter.java]: Why does this class have 2 constructors?

Comment (MarshallAsch) [reply app/src/main/java/ca/marshallasch/veil/tagList/TagListAdapter.java]: The default constructor incase it gets used in the future. its not a bad thing to leave it in. Originally I was using the default constructor then changed to the other one.

Comment (weihanli101) [reply app/src/main/java/ca/marshallasch/veil/tagList/TagListAdapter.java]: Comment it out then. To avoid confusion.

**Pull request merged**